packages/itwinui-css/src/tabs/tabs.scss

```diff
@@ -111,9 +119,19 @@
       height: var(--iui-size-3xs);
     }
   }
+
+  .iui-tabs-actions {
```

Comment:
Thanks! Changed to option 2, since this reduced the specificity.

Reply:
> i could be wrong but i think you actually don't need `@at-root`

I think `@at-root` is required since it is a child of `.iui-tabs-wrapper` & `.iui-horizontal`.
```html
<div class="iui-tabs-wrapper iui-horizontal">
	<ul />
	<div class="iui-tabs-actions" />
	<div class="iui-tabs-content" />
</div>
```

> also we should probably target the tabs wrapper class. `iui-horizontal`/`iui-vertical` are too generic
> and if we want to lower specificity, we can use `:where`

Good point, for safety reasons it makes sense to do this.  Eventually would like to replace these generic classes with `data-iui-orientation` or `aria-orientation`.

Reply:
I implemented the last code block mentioned in https://github.com/iTwin/iTwinUI/pull/860#discussion_r1062703137 but with also including the `&` inside the `:where()` to reduce the specificity even further.

```diff
- :where(.iui-tabs-wrapper.iui-horizontal) & { … }
+ :where(.iui-tabs-wrapper.iui-horizontal &) { … }
```

To confirm, does the second selector above have 0 specificity since everything is in `:where()`?

---

I'm not sure why, but looks like even without `@at-root` works.

I'm curious about the difference between `&` and `#{&}`. Because the generated CSS for both of those are different.

```scss
// Case 1: & (Works)
:where(.iui-tabs-wrapper.iui-vertical &) {} // SCSS
:where(.iui-tabs-wrapper.iui-vertical .iui-tabs-actions) {} // Generated CSS

// Case 2: #{&} (Doesn't work)
:where(.iui-tabs-wrapper.iui-vertical #{&}) {} // SCSS
.iui-tabs-actions :where(.iui-tabs-wrapper.iui-vertical .iui-tabs-actions) {} // Generated CSS

// Case 3: #{&} with @at-root (Works)
@at-root :where(.iui-tabs-wrapper.iui-vertical #{&}) {} // SCSS
:where(.iui-tabs-wrapper.iui-vertical .iui-tabs-actions) {} // Generated CSS
```

For reference, this is how the `.iui-tabs-actions` selector is defined in `tabs/classes.scss`
```scss
.iui-tabs-actions {
  @include iui-tabs-actions;
}
```

Reply:
i don't think we need to make it 0. it's perfectly fine to leave it as `0-1-0` 

the reason it works without `@at-root` is simple: it is officially supported 😄  https://sass-lang.com/documentation/style-rules/parent-selector
![](https://user-images.githubusercontent.com/9084735/210870389-3506efd0-a78c-4d69-8b69-0a1bf3cc9990.png)

case 2 probably doesn't work because it is breaking out of the intended syntax (`.outer-selector &`). when you use `#{}`, it ignores everything and just puts the value there without actually checking what is inside it. my guess: when it first encounters that selector, the sass compiler interprets it as similar to: `:where(.iui-tabs-wrapper.iui-vertical #{something-else})` so it puts the `.iui-tabs-actions` in the beginning as if it was normal nesting.

case 3 works because `@at-root` is specifically used to construct selectors manually, breaking out of implicit nesting.

Reply:
Interesting… it looks like nesting `&` breaks it out of the nesting. 🤔
```scss
@mixin iui-tabs-actions {
  …

  .iui-tabs-wrapper.iui-horizontal & {
    color: red;
  }
}
```

I would have expected:
```css
.iui-tabs-actions .iui-tabs-wrapper.iui-horizontal .iui-tabs-actions { color: red; }
```

But instead got:
```css
.iui-tabs-wrapper.iui-horizontal .iui-tabs-actions { color: red; }
```

Crazy that I've made it this far and never knew this, seems pretty fundemental.

Reply:
> i don't think we need to make it 0. it's perfectly fine to leave it as 0-1-0

Removed `&` from the `:where()` to bring the specificity to 0-1-0 [`418cca5` (#860)](https://github.com/iTwin/iTwinUI/pull/860/commits/418cca506e810c942ea35fffbca106256934e16f)

Reply:
Wow, I just opened `stepper.scss` & removed all `@at-root`s and everything is working as expected. 😦 I bet we can remove a LOT of `@at-root` across our repo now knowing this.